pkg/serviceaccount/claims_test.go

```diff
@@ -182,3 +185,207 @@ func TestClaims(t *testing.T) {
+		},
+		{
+			name:      "pod, deleted now",
+			getter:    fakeGetter{serviceAccount, nil, podDeletedNow},
+			claims:    &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private:   &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Pod: &ref{Name: "podname", UID: "poduid"}, Namespace: "ns"}},
+			expectErr: true,
+		},
+		{
+			name:    "pod, deleted future",
+			getter:  fakeGetter{serviceAccount, nil, podDeletedFuture},
+			claims:  &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private: &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Pod: &ref{Name: "podname", UID: "poduid"}, Namespace: "ns"}},
+		},
+
+		{
+			name:    "secret, valid",
+			getter:  fakeGetter{serviceAccount, secret, nil},
+			claims:  &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private: &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Secret: &ref{Name: "secretname", UID: "secretuid"}, Namespace: "ns"}},
+		},
+		{
+			name:      "secret, missing",
+			getter:    fakeGetter{serviceAccount, nil, nil},
+			claims:    &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private:   &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Secret: &ref{Name: "secretname", UID: "secretuid"}, Namespace: "ns"}},
+			expectErr: true,
+		},
+		{
+			name:      "secret, different uid",
+			getter:    fakeGetter{serviceAccount, secret, nil},
+			claims:    &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private:   &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Secret: &ref{Name: "secretname", UID: "secretuidold"}, Namespace: "ns"}},
+			expectErr: true,
+		},
+		{
+			name:      "secret, deleted past",
+			getter:    fakeGetter{serviceAccount, secretDeletedPast, nil},
+			claims:    &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private:   &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Secret: &ref{Name: "secretname", UID: "secretuid"}, Namespace: "ns"}},
+			expectErr: true,
+		},
+		{
+			name:      "secret, deleted now",
+			getter:    fakeGetter{serviceAccount, secretDeletedNow, nil},
+			claims:    &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private:   &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Secret: &ref{Name: "secretname", UID: "secretuid"}, Namespace: "ns"}},
+			expectErr: true,
+		},
+		{
+			name:    "secret, deleted future",
+			getter:  fakeGetter{serviceAccount, secretDeletedFuture, nil},
+			claims:  &jwt.Claims{Expiry: jwt.NumericDate(nowUnix)},
+			private: &privateClaims{Kubernetes: kubernetes{Svcacct: ref{Name: "saname", UID: "sauid"}, Secret: &ref{Name: "secretname", UID: "secretuid"}, Namespace: "ns"}},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			v := &validator{tc.getter}
+			_, err := v.Validate("", tc.claims, tc.private)
+			if err != nil {
+				if tc.expectErr {
```

Comment:
nit
```
if err != nil && !tc.expectErr {
  t.Fatal(err)
}
```
has better symmetry with the next check.

Reply:
updated